### PR TITLE
Implement get_history function on Device type

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Follow-up of https://github.com/gtk-rs/gdk/pull/314

cc @sdroege @EPashkin @sfanxiang 

PS: Please note that I'll use the CI to confirm that I wrote the code correctly since I don't have the required libs. :)

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
CI passed!

File: src/device.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            let mut n_events = mem::MaybeUninit::uninit();
+            let ret: bool = from_glib(gdk_sys::gdk_device_get_history(
+                self.to_glib_none().0,
+                surface.as_ref().to_glib_none().0,
+                start,
+                stop,
+                &mut events,
+                n_events.as_mut_ptr(),
+            ));
+            if ret {
```
Comment (sfanxiang) [inline src/device.rs]: Shouldn't it be:
```suggestion
            if !ret {
```

**Comment by sfanxiang** [comment]
Otherwise looks good.

File: src/device.rs (hunk 1 above)
Comment (sdroege) [reply src/device.rs]: Yes, see also https://github.com/gtk-rs/gdk/pull/314#discussion_r323085555 :)

Comment (GuillaumeGomez) [reply src/device.rs]: I wonder how I could fail this one too... (bad c/p I guess...)

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/device.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+use std::mem;
+use std::ptr;
+
+impl Device {
+    pub fn get_history<P: IsA<Surface>>(
+        &self,
+        surface: &P,
+        start: u32,
+        stop: u32,
+    ) -> Option<Vec<TimeCoord>> {
```
Comment (sdroege) [inline src/device.rs]: Why `Option<Vec<_>>` instead of just `Vec<_>`? The empty vec is basically like none

Comment (GuillaumeGomez) [reply src/device.rs]: In case the function fails. I find it better to allow users to differentiate an empty return and an erroneous call.

Comment (sdroege) [reply src/device.rs]: It will always return `false` if no value was found. The C function does not allow distinguishing between failure and no items.

Comment (GuillaumeGomez) [reply src/device.rs]: Then it has no sense to return an `Option`. I'll update it.

Comment (sdroege) [reply src/device.rs]: Ok, also needs to be fixed for gdk3 then I guess :)

**Comment by sdroege** [comment]
:+1:

**Comment by GuillaumeGomez** [comment]
cc @sfanxiang

**Comment by sfanxiang** [comment]
:+1:

**Comment by sfanxiang** [comment]
Thanks @GuillaumeGomez!

**Pull request merged**